### PR TITLE
feat: device cert validation

### PR DESCRIPTION
**Pull request by Moopli**
Signed-off-by: Filip Burlacu <filip.burlacu@securekey.com>

### CODE REVIEW

**Comment by Moopli** [comment]
Still needs unit tests and some work on defining the rest api

File: pkg/restapi/operation/operations.go [hunk 1]
```diff
@@ -378,6 +389,71 @@ func (c *Operation) handleBootstrapDataRequest(w http.ResponseWriter, r *http.Re
...
+		cert, e := x509.ParseCertificate(block.Bytes)
+		if e != nil {
+			handleAuthError(w, http.StatusInternalServerError, "failed to parse certificate")
+			return
+		}
+
+		certs = append(certs, cert)
+	}
+
+	if len(certs) == 0 {
```
Comment (llorllale) [inline pkg/restapi/operation/operations.go:425]: this check can be made right after decoding the request into `certHolder`

File: pkg/restapi/operation/operations.go [hunk 2]
```diff
@@ -378,6 +389,71 @@ func (c *Operation) handleBootstrapDataRequest(w http.ResponseWriter, r *http.Re
...
+
+	for _, iCert := range intermediateCerts {
+		intermediatePool.AddCert(iCert)
+	}
+
+	_, err = deviceCert.Verify(x509.VerifyOptions{Intermediates: intermediatePool, Roots: c.deviceRootCerts})
+	if err != nil {
+		// TODO set up proper response
+		// do we redirect?
+		handleAuthError(w, http.StatusTeapot, "TODO response for cert not authenticated")
```
Comment (llorllale) [inline pkg/restapi/operation/operations.go:453]: BadRequest

**Comment by llorllale** [comment]
@Moopli this endpoint must also register and associate the device to the user's profile, therefore needs as input the user's `sub`. We should also accept the AAGUID that identifies the device's model.

**Comment by llorllale** [comment]
Also, endpoint path should be renamed to just `/device`

File: pkg/bootstrap/user/userprofile.go [hunk 3]
```diff
@@ -16,6 +16,7 @@ import (
 // Profile is the user's bootstrap profile.
 type Profile struct {
 	ID                string
+	AAGUID            string
```
Comment (llorllale) [inline pkg/bootstrap/user/userprofile.go:19]: Followup: #58

Comment (Moopli) [reply pkg/bootstrap/user/userprofile.go:19]: I'll add a TODO

**Review by llorllale — APPROVED** [review]

File: pkg/bootstrap/user/userprofile.go (hunk 3 above)
Comment (Moopli) [reply pkg/bootstrap/user/userprofile.go]: done

**Review by talwinder50 — APPROVED** [review]

**Comment by troyronda** [comment]
Codecov should be fixed.

**Pull request merged**